src/main.py

```diff
@@ -32,15 +31,15 @@ async def on_ready() -> None:
 
 @client.event
 async def on_close() -> None:
-    # Unpin current "now playing" message if it exists
-    if bc.now_playing_msg:
-        await discord_utils.try_set_pin(bc.now_playing_msg, False)
-    # Finish current bust (if exists) as if it were stopped
-    await bc.finish(say_goodbye=False)
+    # Finish all running busts on close
+    for _, bc in controllers.items():
+        await bc.finish(say_goodbye=False)
 
 
 @client.event
 async def on_message(message: Message) -> None:
+    global controllers
```

Comment:
Not for this PR, but another thing we could do to get rid of `global`s is to wrap most of the `main.py` login in a `class BustyBot`, which is then instantiated and has, say, a `run` function.

`controllers` would then just be a variable under the `BustyBot` class. So:

```py
class BustyBot:
    controllers: Dict[int, BustController] = {}

    def __init__(self, client: Client) -> "BustController":
        self.client = client

        # Connect to discord
        self.client.run()

    @client.event
    async def on_message(self, message: Message) -> None:
        ...
```